src/utils.py

```diff
@@ -37,6 +39,9 @@
 
 T = TypeVar("T")
 
+TESSERACT_CMD = ["tesseract", "-", "-", "--oem", "1", "--psm", "6"]
```

Comment:
Instead of relying on the Python environment, we should try using `shutil.which` https://docs.python.org/3/library/subprocess.html#subprocess.Popen
> Warning For maximum reliability, use a fully qualified path for the executable. To search for an unqualified name on PATH, use [shutil.which()](https://docs.python.org/3/library/shutil.html#shutil.which). [...]

It can also be made immutable.


```suggestion
import shutil

TESSERACT_CMD = (shutil.which("tesseract"), "-", "-", "--oem", "1", "--psm", "6")
```

Reply:
```suggestion
import shutil

def find_first_valid_path(paths: Iterable[str | None]):
    for path in paths:
        if path and os.path.exists(path):
            return path
    return None


TESSERACT_PATH = find_first_valid_path(
    (
        shutil.which("tesseract"),
        "C:/Program Files/Tesseract-OCR/tesseract.exe",
        os.path.expandvars("%localappdata%/Programs/Tesseract-OCR/tesseract.exe"),
    ),
)
TESSERACT_CMD = (TESSERACT_PATH or "tesseract", "-", "-", "--oem", "1", "--psm", "6")
```